libsweep/src/unix/serial.cc

```diff
@@ -148,7 +153,7 @@ void device_read(device_s serial, void* to, int32_t len) {
         } else {
           throw error{"reading from serial device failed"};
         }
-      } else if(ret == 0){
+      } else if (ret == 0) {
```

Comment:
I think this should check for [feof](http://en.cppreference.com/w/cpp/io/c/feof) (?)